src/View/Helper/FormHelper.php

```diff
@@ -2672,4 +2682,50 @@ public function implementedEvents()
     {
         return [];
     }
+
+    /**
+     * Gets the value sources.
+     *
+     * Returns a list, but at least one item, of valid sources, such as: `'context'`, `'data'` and `'query'`.
+     *
+     * @return array List of value sources.
+     */
+    public function getValuesSources()
+    {
+        return $this->_valuesSources;
+    }
+
+    /**
+     * Sets the value sources.
+     *
+     * Valid values are `'context'`, `'data'` and `'query'`.
+     * You need to supply one valid context or multiple, as a list of strings. Order sets priority.
+     *
+     * @param string|array $sources A string or a list of strings identifying a source.
+     * @return $this
+     */
+    public function setValuesSources($sources)
+    {
+        $this->_valuesSources = array_values(array_intersect((array)$sources, ['context', 'data', 'query']));
+
+        return $this;
+    }
+
+    /**
+     * Gets a single field value from the sources available.
+     *
+     * @param string $fieldname The fieldname to fetch the value for.
+     * @return string Field value derived from sources.
```

Comment:
Should be `string|null` as the bottom of the method can return a `null` implicitly.